spec/electric_slide/call_queue_spec.rb

```diff
@@ -138,6 +145,34 @@
           end
         end
       end
+
+      context "when the agent's and caller's calls are joined" do
+        before do
+          queued_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+          agent_call << Punchblock::Event::Joined.new(timestamp: connected_time)
+        end
+
+        it "records the connection time in the :electric_slide_connected_at call variable on the queued call" do
+          expect(queued_call[:electric_slide_connected_at]).to eq(connected_time)
+        end
+      end
+
+      context "when the agent's and caller's calls are not joined" do
```

Comment:
I get that. What you have now is effectively this:

```
context "when the calls are not joined"
context "when the calls are joined"
context "when the calls are not joined"
```

I'm wondering why you don't include the last test in the first group.